Fork_Maintenance.md

```diff
@@ -0,0 +1,263 @@
+
+*(Example - Adapt based on actual build commands, assuming build context is repo root)*
+```bash
+# Example: Build triggered by CI on release/stable-with-patches
+git checkout release/stable-with-patches
+MAIN_COMMIT_BASE=$(git log -n 1 --pretty=%H main) # Get the latest main commit integrated
+BUILD_TAG="stable-$(date +%Y%m%d)-$(git rev-parse --short HEAD)-main-${MAIN_COMMIT_BASE:0:7}"
+
+docker build -t your-registry/openhands-runtime:${BUILD_TAG} -f path/to/runtime/Dockerfile .
+docker build -t your-registry/openhands-agent:${BUILD_TAG} -f path/to/agent/Dockerfile .
+
+# Optionally, also tag as 'latest-stable' or similar
+docker tag your-registry/openhands-runtime:${BUILD_TAG} your-registry/openhands-runtime:latest-stable
+docker tag your-registry/openhands-agent:${BUILD_TAG} your-registry/openhands-agent:latest-stable
+```
+
+**b. Tagging Strategy:**
+
+*   **Primary Tag:** Use a descriptive tag that indicates the source branch, date, commit SHA of the release branch, and potentially the `main` branch commit it's based on (see example above). E.g., `stable-YYYYMMDD-<release-sha>-main-<main-sha>`.
+*   **Floating Tag (Optional):** Maintain a floating tag like `latest-stable` that points to the most recent successful build from `release/stable-with-patches`. This simplifies configuration for environments that should always use the latest stable version.
+
+**c. Distribution:**
+
+Push the tagged images (both specific and floating, if used) to your container registry (e.g., Docker Hub, GHCR, private registry).
+
+```bash
+# Example: Push images (adapt tags)
+docker push your-registry/openhands-runtime:${BUILD_TAG}
+docker push your-registry/openhands-agent:${BUILD_TAG}
+docker push your-registry/openhands-runtime:latest-stable
+docker push your-registry/openhands-agent:latest-stable
+```
+
+**d. Configuration:**
+
+Update your OpenHands configurations (environment variables, config files, deployment manifests) to use your custom image tags. Prefer using a specific build tag for production/critical environments and the floating tag (`latest-stable`) for development or less critical deployments.
+
+*   `RUNTIME_IMAGE=your-registry/openhands-runtime:latest-stable` (or specific tag)
+*   `AGENT_IMAGE=your-registry/openhands-agent:latest-stable` (or specific tag)
+
+Ensure these configurations are applied consistently across all environments (local dev, CI/CD, resolver flows).
+
+**The CI/CD pipeline for `release/stable-with-patches` handles the build, tag, and push process automatically.** Configuration updates might be manual or automated depending on your deployment strategy.
+
+## 3. Upstream Synchronization and Integration
```

Comment:
As I mentioned above, I envision this section to be an openhands task that runs regularly to generate a PR that we review.